src/main/java/org/embulk/output/kintone/KintoneColumnVisitor.java

```diff
@@ -53,156 +50,219 @@ public void setUpdateKey(UpdateKey updateKey) {
+    } else if (pageReader.isNull(column)) {
+      setBoolean(type, fieldCode, updateKey, false, option);
+    } else {
+      setBoolean(type, fieldCode, updateKey, pageReader.getBoolean(column), option);
     }
-    record.putField(fieldCode, fieldValue);
   }
 
-  private void setTimestampValue(String fieldCode, Instant instant, ZoneId zoneId, FieldType type) {
-    FieldValue fieldValue = null;
-    ZonedDateTime datetime = instant.atZone(zoneId);
-    switch (type) {
-      case DATE:
-        fieldValue = new DateFieldValue(datetime.toLocalDate());
-        break;
-      case DATETIME:
-        fieldValue = new DateTimeFieldValue(datetime);
+  @Override
+  public void longColumn(Column column) {
+    if (isIgnoreNull(column)) {
+      return;
+    }
+    KintoneColumnOption option = getOption(column);
+    UpdateKey updateKey = getUpdateKey(column);
+    KintoneColumnType type = KintoneColumnType.getType(option, KintoneColumnType.NUMBER);
+    String fieldCode = getFieldCode(column);
+    if (isPreferNull(column)) {
+      setNull(type, fieldCode, updateKey);
+    } else if (pageReader.isNull(column)) {
+      setLong(type, fieldCode, updateKey, 0, option);
+    } else {
+      setLong(type, fieldCode, updateKey, pageReader.getLong(column), option);
     }
-    record.putField(fieldCode, fieldValue);
   }
 
-  private void setCheckBoxValue(String fieldCode, Object value, String valueSeparator) {
-    String str = String.valueOf(value);
-    CheckBoxFieldValue checkBoxFieldValue = new CheckBoxFieldValue();
-
-    if (str != null && !str.equals("")) {
-      List<String> values = Arrays.asList(str.split(valueSeparator, 0));
-      checkBoxFieldValue = new CheckBoxFieldValue(values);
+  @Override
+  public void doubleColumn(Column column) {
+    if (isIgnoreNull(column)) {
+      return;
+    }
+    KintoneColumnOption option = getOption(column);
+    UpdateKey updateKey = getUpdateKey(column);
+    KintoneColumnType type = KintoneColumnType.getType(option, KintoneColumnType.NUMBER);
+    String fieldCode = getFieldCode(column);
+    if (isPreferNull(column)) {
+      setNull(type, fieldCode, updateKey);
+    } else if (pageReader.isNull(column)) {
+      setDouble(type, fieldCode, updateKey, 0, option);
+    } else {
+      setDouble(type, fieldCode, updateKey, pageReader.getDouble(column), option);
     }
-    record.putField(fieldCode, checkBoxFieldValue);
   }
 
-  private FieldType getType(Column column, FieldType defaultType) {
-    KintoneColumnOption option = columnOptions.get(column.getName());
-    if (option == null) {
-      return defaultType;
+  @Override
+  public void stringColumn(Column column) {
+    if (isIgnoreNull(column)) {
+      return;
+    }
+    KintoneColumnOption option = getOption(column);
+    UpdateKey updateKey = getUpdateKey(column);
+    KintoneColumnType defaultType =
+        updateKey != null ? KintoneColumnType.SINGLE_LINE_TEXT : KintoneColumnType.MULTI_LINE_TEXT;
```

Comment:
If the column is an update key, the default type is SINGLE_LINE_TEXT instead of MULTI_LINE_TEXT.